core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -4385,15 +4451,39 @@ class KafkaApisTest {
         .setGroupId("group-3")
         .setErrorCode(Errors.INVALID_GROUP_ID.code)
 
-      val expectedOffsetFetchResponse = new OffsetFetchResponseData()
-        .setGroups(List(group1Response, group2Response, group3Response).asJava)
+      val group4Response = new OffsetFetchResponseData.OffsetFetchResponseGroup()
+        .setGroupId("group-4")
+        .setErrorCode(Errors.INVALID_GROUP_ID.code)
+
+      val group5Response = new OffsetFetchResponseData.OffsetFetchResponseGroup()
```

Comment:
I don't fully understand this test case either. My understanding is that we could have a weird response only in one case: the coordinator returns a response with an error and the kafka apis adds unauthorized topics to it. Did I get it right? If so, it may be better to actually add a new (and separate) test case for this one.

Reply:
Yeah, you're right. Let me write it to a separate method. We probably don't need this group 5